quarkchain/cluster/shard_state.py

```diff
@@ -1950,6 +1950,73 @@ def get_root_chain_stakes(
         signer = output[32 + 12 :]
         return stakes, signer
 
+    def get_gas_utility_info(self, token_id: int, evm_state) -> (int, int):
```

Comment:
according to your PR #784, this method is called on `State` class in `quarkchain/evm/state.py`, not `ShardState`